services/accounts/accounts.go

```diff
@@ -197,3 +290,73 @@ func (api *API) verifyPassword(password string) error {
 	_, err = api.manager.VerifyAccountPassword(api.config.KeyStoreDir, address.Hex(), password)
 	return err
 }
+
+func (api *API) GetAddressListForPath(password string, derivedFrom string, path string, pageSize int, pageNumber int) ([]*Address, error) {
+	derivedAddresses := make([]*Address, 0)
+
+	info, err := api.manager.AccountsGenerator().LoadAccount(derivedFrom, password)
+	if err != nil {
+		return nil, err
+	}
+
+	if pageNumber <= 0 || pageSize <= 0 {
+		return nil, fmt.Errorf("pageSize and pageNumber should be greater than 0")
+	}
+
+	var startIndex = (pageNumber - 1) * pageSize
+	var endIndex = pageNumber * pageSize
+
+	for i := startIndex; i < endIndex; i++ {
+		derivedPath := fmt.Sprint(path, "/", i)
+
+		infos, err := api.manager.AccountsGenerator().DeriveAddresses(info.ID, []string{derivedPath})
+		if err != nil {
+			return nil, err
+		}
+
+		address := &Address{
+			Address:     common.HexToAddress(infos[derivedPath].Address),
+			Path:        derivedPath,
+			HasActivity: false,
+		}
+
+		derivedAddresses = append(derivedAddresses, address)
+	}
+
+	return derivedAddresses, nil
+}
+
+func (api *API) GetAddressListForMenominicWithPath(mnemonic string, path string, pageSize int, pageNumber int) ([]*Address, error) {
```

Comment:
Same suggestion:
GetDerivedAddressesForMnemonicAndPath

Also this function seems to duplicate code